src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,883 @@
+      | None -> (* the variable is equal to a constant *) ()
+      | Some ref_var ->
+        if ref_var <> var then ()
+        else
+          (* x_i is the reference variable of its connected component *)
+          let dim_of_var = Some var in
+          let connected_component = find_vars_in_the_connected_component d dim_of_var in
+          if length connected_component = 1 
+          then ()   (* x_i is the only element of its connected component *) 
+          else
+            (* x_i is the reference variable -> we need to find a new reference variable *)
+            let var_least_index = Option.get @@ find_var_in_the_connected_component_with_least_index connected_component ref_var in
+            let (_, off) = d.(var_least_index) in 
+            iteri (fun _ x -> let (_, off2) = d.(x) in if x <> ref_var then d.(x) <- (Some var_least_index, Z.(off2 - off))) connected_component;
+    end
+
+  (* Forget information about variable i but not in-place *)
+  let reduce_col m j = 
+    let copy = copy m in
+    reduce_col_with copy j;
+    copy 
+
+  let remove_zero_rows t = t 
+
+end
+
+
+module V = RelationDomain.V
+
+(** It defines the type t of the affine equality domain (a struct that contains an optional matrix and an apron environment) and provides the functions needed for handling variables (which are defined by RelationDomain.D2) such as add_vars remove_vars.
+    Furthermore, it provides the function get_coeff_vec that parses an apron expression into a vector of coefficients if the apron expression has an affine form. *)
+module VarManagement =
+struct
+  module EArray = EqualitiesArray
+  include SharedFunctions.VarManagementOps (EArray)
+
+
+  let size t = match t.d with
+    | None -> 0 
+    | Some d -> EArray.length d
+
+  (* Returns the constant value of a variable, 
+     if we know the constant value of this variable.
+     Else it returns None. *)
+  let get_variable_value_if_it_is_a_constant t var =
+    let get_constant (var, off) = match var with
+      | None -> Some off
+      | _ -> None
+    in
+    Option.bind t.d (fun d -> get_constant d.(var))
+
+  let get_coeff_vec (t: t) texp =
+    (*Parses a Texpr to obtain a (coefficient, variable) pair list to repr. a sum of a variables that have a coefficient. If variable is None, the coefficient represents a constant offset. 
+    *)
+    let open Apron.Texpr1 in
+    let exception NotLinearExpr in
+    let exception NotIntegerOffset in
+    let negate coeff_var_list = List.map (fun (coeff, var) -> (Z.(-coeff), var)) coeff_var_list in
+    let multiply_with_Z number coeff_var_list = 
+      List.map (fun (coeff, var) -> (Z.(number * coeff, var))) coeff_var_list in
+    let multiply a b = 
+      (* if one of them is a constant, then multiply. Otherwise, the expression is not linear*)
+      if List.compare_length_with a 1 = 0 then
+        match List.nth a 0 with
+        | (a_coeff, None) -> multiply_with_Z a_coeff b
+        | _ -> raise NotLinearExpr
```

Comment:
Just one observation based on a fix commit. This is the perfect opportunity to use pattern matching:
```suggestion
      match a with
      | [(a_coeff, None)] -> multiply_with_Z a_coeff b
      | _ -> raise NotLinearExpr
```